Create Prompts List Activity and its navigation action

metalwihen opened the pull request:
Fixes #17125

**NOTE: This PR should not be merged into `trunk` yet. This is one small part of the task.** 

**To test**
- Menu options under Prompts card
- Clicking "View more prompts" should open activity, PromptsActivity

| Image  | Description |
| ------------- | ------------- |
| ![image](https://user-images.githubusercontent.com/2140539/193133604-443b3787-bcc1-4ff7-aaa6-74e4055d8217.png)  | Screenshot of the Home screen once the site is selected. A new menu option has been added to the Prompts card, "View more options" which will lead to a newly created Prompts List Activity (currently blank)  |


## Regression Notes
**1. Potential unintended areas of impact**
  - Changes to test classes depending on Prompt Dashboard card

**2. What I did to test those areas of impact (or what existing automated tests I relied on)**
  - Manual test to ensure the click listener for menu option "View More Prompts" is working as expected.
  - Manual test to ensure the Prompts List Activity opens on clicking "View More Prompts" screen (currently blank)
  - Manual test to ensure that the other menu options ("Skip", "Learn more") work as before.
  - Run existing unit tests to ensure there is no regression

**3. What automated tests I added (or what prevented me from doing so)**
  - Unit tests to ensure the click listener for "View More Prompts" is handled in ViewModel.


## PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

metalwihen committed:
Add Blogging Prompts List Activity

metalwihen committed:
Add 'View More' menu option to Prompts Dashboard Card

RenanLukas commented on WordPress/src/main/java/org/wordpress/android/ui/bloggingprompts/BloggingPromptsActivity.kt:
```diff
@@ -0,0 +1,74 @@
+import org.wordpress.android.ui.LocaleAwareActivity
+import org.wordpress.android.util.AppLog
+
+class BloggingPromptsActivity : LocaleAwareActivity() {
```
I was wondering if we really need to inherit from `LocaleAwareActivity` in this case 🤔 
Do we use Locale somehow in this context?

metalwihen replied:
**My reasoning for using it was:**
- All the activities I used as a reference (PostListActivity, StatsActivity, PagesActivity) are extending from LocaleAwareActivity. It seemed like a norm to be followed. 
- May be missing some context here but my understanding is that if the locale was changed, the LocaleAwareActivity class would help handle the locale changes at activity level. 

**That said**, to answer the question of whether Locale is needed in this context. I guess it depends on how locale change is handled in the Wordpress app. Is there a feature to change locale, in-app, programmatically? If not, we can set the expectations for the user to restart the activity to see the locale changes and remove the need for extending `LocaleAwareActivity`.

metalwihen replied:
I've added a separate comment to remove the dependency on LocaleAwareActivity.

RenanLukas replied:
> I guess it depends on how locale change is handled in the Wordpress app. Is there a feature to change locale, in-app, programmatically?

Good point, @metalwihen .

I believe inheriting from `LocaleAwareActivity` is needed to get the correct Locale while picking which translations to use. I'm fine with using `AppCompatActivity` now and maybe in the UI PR we can test if the translations work as expected while changing the device Locale with this screen opened.

RenanLukas commented on WordPress/src/main/java/org/wordpress/android/ui/bloggingprompts/BloggingPromptsActivity.kt:
```diff
@@ -0,0 +1,74 @@
+        return super.onOptionsItemSelected(item)
+    }
+
+    override fun onNewIntent(intent: Intent) {
```
Considering we have `standard` launch mode for  `BloggingPromptsActivity`, I was wondering if the `onNewIntent` method would ever be called 🤔

metalwihen replied:
True. I'll remove this method.

metalwihen replied:
Done.

RenanLukas commented on WordPress/src/main/java/org/wordpress/android/ui/bloggingprompts/BloggingPromptsActivity.kt:
```diff
@@ -0,0 +1,74 @@
+
+    override fun onNewIntent(intent: Intent) {
+        if (!intent.hasExtra(WordPress.SITE)) {
+            AppLog.e(AppLog.T.MY_SITE_DASHBOARD, "BloggingPromptsActivity started without a site.")
```
[nitpick]: do you think it would make sense to have something like `"${BloggingPromptsActivity::class.simpleName} started without a site."` instead to make sure that if the class is renamed, the message will still make sense?

metalwihen replied:
Makes sense. Will fix the message

metalwihen replied:
Done. While this particular message was removed with the unused method, I did update the copy of the other error message in the class.

RenanLukas commented on WordPress/src/main/res/layout/blogging_prompts_list_activity.xml:
```diff
@@ -0,0 +1,12 @@
+        android:id="@+id/fragment_container"
+        android:layout_width="match_parent"
+        android:layout_height="match_parent" />
+    <!-- TODO: android:name -->
```
Would it make sense to remove this `TODO` before merging?

metalwihen replied:
Will do.

metalwihen replied:
Done

RenanLukas reviewed (commented):
Hey @metalwihen 👋 
Thank you for the PR! I've finished my initial review and left a couple of questions 🙂

metalwihen committed:
Apply PR suggestions

- Remove TODO comments
- Remove onNewIntent() from activity whose launchMode is standard
- Fix copy of error messages

metalwihen committed:
Apply PR suggestions

- Remove LocaleAwareActivity dependency
- Fix detekt error, MaxLineLength

RenanLukas commented on WordPress/src/main/java/org/wordpress/android/ui/bloggingprompts/BloggingPromptsActivity.kt:
```diff
@@ -0,0 +1,54 @@
+
+        site = if (savedInstanceState == null) {
+            checkNotNull(intent.getSerializableExtra(WordPress.SITE) as? SiteModel) {
+                "${WordPress.SITE} argument cannot be null, check the PendingIntent starting ${BloggingPromptsActivity::class.simpleName}"
```
Hey @metalwihen , thanks for the changes!

`detekt` is giving me a `MaxLineLength` error in this line. Could you please have a look?

metalwihen replied:
Fixed

RenanLukas reviewed (approved):
Thanks for the changes!
LGTM :shipit:
